flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/MySqlToElasticsearchE2eITCase.java

```diff
@@ -0,0 +1,231 @@
+                        + "  parallelism: 1",
+                MYSQL_TEST_USER,
+                MYSQL_TEST_PASSWORD,
+                mysqlInventoryDatabase.getDatabaseName(),
+                getElasticsearchVersion().split("\\.")[0]  // Use major version number
+        );
+
+        Path mysqlCdcJar = TestUtils.getResource("mysql-cdc-pipeline-connector.jar");
+        Path elasticsearchCdcConnector = TestUtils.getResource("flink-cdc-pipeline-connector-elasticsearch-3.2-SNAPSHOT.jar");
+        Path mysqlDriverJar = TestUtils.getResource("mysql-driver.jar");
+
+        submitPipelineJob(pipelineJob, mysqlCdcJar, elasticsearchCdcConnector, mysqlDriverJar);
+        waitUntilJobRunning(Duration.ofSeconds(30));
+        LOG.info("Pipeline job is running");
+
+        // Wait for data to be written
+        Thread.sleep(20000);
+        validateElasticsearchContent("customers");
+        validateElasticsearchContent("products");
+
+        LOG.info("Begin incremental reading stage.");
+        String mysqlJdbcUrl = String.format(
+                "jdbc:mysql://%s:%s/%s",
+                MYSQL.getHost(),
+                MYSQL.getDatabasePort(),
+                mysqlInventoryDatabase.getDatabaseName()
+        );
+
+        try (Connection conn = DriverManager.getConnection(mysqlJdbcUrl, MYSQL_TEST_USER, MYSQL_TEST_PASSWORD);
+             Statement stat = conn.createStatement()) {
+
+            stat.execute("INSERT INTO products VALUES (default,'jacket','water resistent white wind breaker',0.2, null, null, null);");
+
+            // Wait for data to be written
+            Thread.sleep(20000);
+            validateElasticsearchContent("products");
+
+            stat.execute("UPDATE products SET description='18oz carpenter hammer' WHERE id=106;");
+            stat.execute("UPDATE products SET weight='5.1' WHERE id=107;");
+
+            stat.execute("ALTER TABLE products DROP COLUMN point_c;");
+            stat.execute("DELETE FROM products WHERE id=101;");
+
+            stat.execute("INSERT INTO products VALUES (default,'scooter', null, 2.14, null, null);");
+
+            // Wait for data to be written
+            Thread.sleep(20000);
+        } catch (SQLException e) {
+            LOG.error("Update table for CDC failed.", e);
+            throw e;
+        }
+
+        validateElasticsearchContent("products");
+    }
+
+    private void validateElasticsearchContent(String tableName) throws Exception {
+        String indexName = String.format("%s.%s", mysqlInventoryDatabase.getDatabaseName(), tableName);
+        int hostPort = elasticsearch.getMappedPort(9200);
+        String url = String.format("http://localhost:%d/%s/_search?pretty", hostPort, indexName);
+        String curlCommand = String.format("curl -X GET '%s'", url);
+
+        ProcessBuilder processBuilder = new ProcessBuilder("curl", "-X", "GET", url);
+        Process process = processBuilder.start();
+        int exitCode = process.waitFor();
```

Comment:
`curl` is not generally available on all platforms (like Windows), even some Linux distro doesn't package it by default. Please consider using Java HTTP libraries to send request instead.